modules/dreamview/backend/simulation_world/simulation_world_service.cc

```diff
@@ -629,6 +631,24 @@ void SimulationWorldService::SetObstacleSensorMeasurements(
   }
 }
 
+void SimulationWorldService::SetObstacleSource(
+    const apollo::perception::PerceptionObstacle &obstacle,
+    Object *world_object) {
+  if (world_object == nullptr || !obstacle.has_source()) {
+    return;
+  }
+  const PerceptionObstacle::Source obstacle_source = obstacle.source();
+  world_object->set_source(obstacle_source);
+  if (obstacle_source == PerceptionObstacle::V2X) {
+    V2XInformation *v2x_info = world_object->mutable_v2x_info();
+    v2x_info->clear_v2x_type();
```

Comment:
we don't need to clear v2x_type every time even when there's no PerceptionObstacle::V2X?

Reply:
我的理解是没有source就没有v2xinfo，就不需要清理，有source的情况下有v2xinfo需要清理

Reply:
isn't it possible to have source from the previous frame but not for the current one. In that case, we need to clear the data, otherwise, previous frame's remain.